### PR TITLE
zfs: use meta.broken instead of throw when incompatible with kernel

### PR DESCRIPTION
**Pull request by Atemu**
###### Motivation for this change
throw aborts eval when the package is touched in inappropriate places, see
https://github.com/NixOS/nixpkgs/issues/109001 for an adverse instance of that.

It now behaves like a regular broken package when it's, you know, broken.

It also still prints the helpful incompatibility notice.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by NeQuissimus — APPROVED** [review]
LGTM. I do wonder why `meta.broken` was not used originally

**Review by thiagokokada — APPROVED** [review]
Diff LGTM.

**Review by hmenke — APPROVED** [review]

**Comment by Atemu** [comment]
Whoops, accidentally tried to merge staging into master; sorry for the ping.

Merge conflict fixed.

**Comment by Mic92** [comment]
safe for backporting.

File: pkgs/os-specific/linux/zfs/default.nix [hunk 1]
```diff
@@ -174,6 +168,13 @@ let
         license = licenses.cddl;
         platforms = platforms.linux;
         maintainers = with maintainers; [ hmenke jcumming jonringer wizeman fpletz globin mic92 ];
+        broken = if
+          buildKernel && (incompatibleKernelVersion != null) && versionAtLeast kernel.version incompatibleKernelVersion
```
Comment (jonringer) [inline pkgs/os-specific/linux/zfs/default.nix:172]:
```suggestion
          buildKernel && (incompatibleKernelVersion != null) && kernel.kernelAtLeast incompatibleKernelVersion
```

**Review by jonringer — COMMENTED** [review]
I guess I'm late

File: pkgs/os-specific/linux/zfs/default.nix (hunk 1 above)
Comment (Atemu) [reply pkgs/os-specific/linux/zfs/default.nix:172]: LGTM, feel free to just push that.

Could you also do the backport to stable (staging?)? You'd probably be the one to merge my PR for that anyways ;)

Comment (Atemu) [reply pkgs/os-specific/linux/zfs/default.nix:172]: Although hold on, have we foreward-ported kernelAtLeast yet? You created that in staging IIRC.

Comment (jonringer) [reply pkgs/os-specific/linux/zfs/default.nix:172]: I should be there, I needed it to properly mark failing module builds

Comment (jonringer) [reply pkgs/os-specific/linux/zfs/default.nix:172]:
```
[11:42:55] jon@jon-desktop /home/jon/projects/nixpkgs (release-20.09)
$ git pull
Already up to date.
[11:42:59] jon@jon-desktop /home/jon/projects/nixpkgs (release-20.09)
$ nix eval -f . linuxPackages.kernel.kernelAtLeast
<LAMBDA>
```

**Pull request merged**